Closes #3184: Download Limit Support (#3311)
nextcloud/ios: README.md

## Testing

**Note: If a unit or integration test exclusively uses and tests NextcloudKit functions and components, then add that test to the NextcloudKit project instead. NextcloudKit is used in many other projects as an API, and it's better if such tests are located there.**

### Unit Tests

There are currently no prerequisites for unit testing. Mock everything which not needed. 

### Integration Tests

In `TestConstants.swift` you must specify your credentials for the server you want to test against.
[The Nextcloud Docker image](https://github.com/nextcloud/docker) maintained by the community is an easy and fast way to spin up a Nextcloud server to test against. 
[This shallow CI server image](https://github.com/nextcloud/docker-ci/pkgs/container/continuous-integration-shallow-server) is also used by the Android project for automated tests in the GitHub pipeline.
This latter one will be referred to from here on.

> With the following commands, you have will set up a new Nextcloud available on local HTTP port 8080, once the Docker image was initially pulled.

Reviewer: ```suggestion
With the following commands, you would set up a new Nextcloud Docker instance available on local HTTP port 8080.
```

```sh
#!/usr/bin/env zsh

# The fixed Docker container name for the Nextcloud server.
CONTAINER_NAME="xcode-ui-test-server"

# Check for environment variable of server version to use including a default value.
SERVER_VERSION=${SERVER_VERSION:-"stable30"}

# Launch Nextcloud Server in Docker Container.
docker run \
    --detach \
    --name $CONTAINER_NAME \
    --publish 8080:80 \
    ghcr.io/nextcloud/continuous-integration-shallow-server:latest

# Enable File Download Limit App.
docker exec $CONTAINER_NAME su www-data -c "git clone --depth 1 -b $SERVER_VERSION https://github.com/nextcloud/files_downloadlimit.git /var/www/html/apps/files_downloadlimit/"
docker exec $CONTAINER_NAME su www-data -c "php /var/www/html/occ app:enable files_downloadlimit"

> ```

Reviewer: As discussed you can just create this .sh file and include it in Tests. And you can also update the README explaining how to run it etc.
